src/components/checkbox/checkbox.js

```diff
@@ -156,11 +175,18 @@ function MdCheckboxDirective(inputDirective, $mdAria, $mdConstant, $mdTheming, $
           listener(ev);
         }
       }
+
       function listener(ev) {
         if (element[0].hasAttribute('disabled')) {
           return;
         }
 
+        if (indeterminateStateEnabled) {
+          if (attr.indeterminateClick) {
+            scope.$apply(element.attr('indeterminate-click'));
```

Comment:
I ran into some issues with the Speed Dial when I was using `$apply` instead of `$evalAsync` because there were cases when the code could be called during an existing digest and throw an error.

If I recall, this occurred when a user added a custom `ng-click` to the component, so you may want to double-check that case.